File: sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def dfs(self,root,path):
        if root == None:
            return
        
        if root.left == None and root.right == None:
            path = path+str(root.val)
            #print(path)
            self.res += int(path)
        
        self.dfs(root.left,path+str(root.val))
        self.dfs(root.right,path+str(root.val))
        return
    
    
    def sumNumbers(self, root: TreeNode) -> int:
        res = []
        def recur(root,path):
            if root == None:
                return
            
            recur(root.left,path+str(root.val))
            recur(root.right,path+str(root.val))
            
            if root.left == None and root.right == None:
                res.append(path+str(root.val))
        
            return
        
        #recur(root,"")
        #res = [int(x) for x in res]
        #res = sum(res)
        #return res
        
        self.res = 0
        self.dfs(root,"")
        return self.res
        

File: sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int rootToLeaf(TreeNode root, int[] totalSum, String value){
        
        if(root == null){
            return 0;
        }
        
        if(root.left == null && root.right == null){
            //System.out.println(value+"leaf");
            totalSum[0] += Integer.parseInt(value+ root.val) ;
        }
        rootToLeaf(root.left,totalSum,value + Integer.toString(root.val));
        rootToLeaf(root.right,totalSum,value + Integer.toString(root.val));
        return totalSum[0];
    }
    public int sumNumbers(TreeNode root) {
        
        int[] totalSum = {0};
        int sum = rootToLeaf(root,totalSum,"");
        return sum;
    }
}